locations/spiders/rural_king_us.py

```
@@ -0,0 +1,316 @@
+                "phone": store.get("telephone1", "").strip(),
+                "website": website,
+                "lat": float(store.get("latitude")),
+                "lon": float(store.get("longitude")),
+            }
+
+            # Extract email if available
+            for attribute in store.get("Attribute", []):
+                if attribute.get("name") == "Email":
+                    properties["email"] = attribute.get("value")
+                    break
+
+            # Get store image if available
+            for description in store.get("Description", []):
+                if "thumbnail" in description:
+                    properties["image"] = description.get("thumbnail")
+                    break
+
+            # Add hours
+            hours = self.parse_hours(store)
+            if hours:
+                properties["opening_hours"] = hours
+
+            # Create the feature
+            item = Feature(**properties)
+
+            # Apply shop category - Rural King is an agrarian/farm supply store
+            apply_category(Categories.SHOP_AGRARIAN, item)
+
+            # Add store services based on what we found on the store page
+            extras = item.get("extras", {})
+
+            # Apply services only if they actually exist for this store
+            if "Bulk Propane Tanks Filled & Sold Here" in services:
+                apply_yes_no(Fuel.PROPANE, item, True)
+
+            if "RKGuns" in services:
+                extras["shop"] = "guns"
+
+            if "In-Store Pickup" in services:
+                extras["pickup"] = "yes"
+
+            if "Ship to Store" in services:
+                extras["delivery:store"] = "yes"
+
+            # Store the raw services list for reference
+            extras["store_services"] = services
+
+            item["extras"] = extras
+
+            return item
+        except Exception as e:
+            self.logger.error(f"Error creating store item: {str(e)}")
+            return None
```

Generally I prefer to not wrap parsing functions in try/catch and let any errors bubble up to cause the spider to fail.